packages/javascript/bh-shared-ui/src/utils/theme.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright 2024 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+import { clsx, type ClassValue } from 'clsx';
+import { twMerge } from 'tailwind-merge';
+
+/**
+ * This function sets the name of our current theme mode as a class on the html document root. This will ensure the correct styles are applied to components attached elsewhere in the DOM, such as modals and popover menus.
+ *
+ * @param enabled - set to 'true' if dark mode is currently enabled
+ *
+ * @returns the name of the currently set mode as a string
+ */
+export const toggleModeClassOnRoot = (enabled: boolean) => {
+    const root = window.document.documentElement;
+    const mode = enabled ? 'dark' : 'light';
+
+    root.classList.remove('dark', 'light');
+    root.classList.add(mode);
+
+    return mode;
+};
```

Comment:
To me it makes more sense if this function applies a specific class to the root element. enabled = dark and disabled = light is not straightforward to me